### PR TITLE
Auto-Register Schema Config Check in Json Serializers

### PR DESCRIPTION
**Pull request by nickghardwick**
This PR adds a check in the Java and .NET serializers using schema registry for the auto-register schema configuration. Previously the code always registered new schemas upon serialization; this fix allows the user to set the auto-register behavior in the serializer configuration before running the serializers.

### CODE REVIEW

File: csharp/json/src/KafkaJsonDeserializer.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public KafkaJsonDeserializer(string schemaRegistryUrl, TokenCredential credentia
             {
                 Diagnostics =
                 {
-                    ApplicationId = "azsdk-net-KafkaJsonDeserializer/1.0"
```
Comment (hmlam) [inline csharp/json/src/KafkaJsonDeserializer.cs:37]: Is this change intentional? if so why do we need this?

File: csharp/json/src/KafkaJsonSerializer.cs [hunk 2]
```diff
@@ -26,17 +26,19 @@ public class KafkaJsonSerializer<T> : ISerializer<T>
...
         readonly string schemaGroup;
 
-        public KafkaJsonSerializer(string schemaRegistryUrl, TokenCredential credential, string schemaGroup)
+        public KafkaJsonSerializer(string schemaRegistryUrl, TokenCredential credential, string schemaGroup, bool autoRegisterSchemas = false)
         {
             this.schemaRegistryClient = new SchemaRegistryClient(schemaRegistryUrl, credential, new SchemaRegistryClientOptions
             {
                 Diagnostics =
                 {
-                    ApplicationId = "azsdk-net-KafkaJsonSerializer/1.0"
```
Comment (hmlam) [inline csharp/json/src/KafkaJsonSerializer.cs:37]: same comment as the deserializer

**Review by hmlam — CHANGES_REQUESTED** [review]

File: csharp/json/src/KafkaJsonDeserializer.cs (hunk 1 above)
Comment (nickghardwick) [reply csharp/json/src/KafkaJsonDeserializer.cs:37]: When testing I found that the serializer throws an error when the ApplicationId is longer than 25 characters

**Comment by AndrijaRaguz** [comment]
Hello! Do you intend to add a fix that allows user to set the auto-register behavior in the AvroConverter configuration?
Because autoregistration is currently hardcoded to true in the AvroConverter serializer and that is why schemas are automatically registered in Kafka Connect scenarios.

**Comment by hmlam** [comment]
> > Hello! Do you intend to add a fix that allows user to set the auto-register behavior in the AvroConverter configuration? Because autoregistration is currently hardcoded to true in the AvroConverter serializer and that is why schemas are automatically registered in Kafka Connect scenarios.
> 
> @hmlam Looks like this same fix can be made in the Kakfa Connect scenario as well, I can add that here or in a separate PR for tracking (since this fix is tied to a specific CRI)

we should just fix all in one PR since they are related.

**Review by hmlam — APPROVED** [review]
